packages/webdriver/src/index.js

```diff
@@ -66,17 +66,12 @@ export default class WebDriver {
      * allows user to attach to existing sessions
      */
     static attachToSession (options = {}, modifier, proto = {}, commandWrapper) {
-        if (typeof options.sessionId !== 'string') {
-            throw new Error('sessionId is required to attach to existing session')
-        }
-
-        logger.setLevel('webdriver', options.logLevel)
+        const params = validateConfig(SESSION_DEFAULTS, options)
```

Comment:
Can we get this working without validating the session again? It should not be required if the session was valid before.

Reply:
Unless there is a way to distinguish between prior validated sessions and "raw" session responses which are passed to create new webdriver(io) instances (e.g. the snippet I've commented above).
Alternative would be not to expose `attach` or `WebDriver.attachToSession`.
Otherwise default values should be specified, this is not much different from piping default values like it was done before.

Reply:
As mentioned in the comment above, there are cases where the sessions might be valid before, but wrapped in another class (e.g. `MultiRemote`), which simply does not have these "default" values at their disposal.

This way we always have the required values when using an external session or a wrapper for multiple sessions.

Reply:
Ultimately I would love to have such an interface:

```js
const driver = await remote({ ... })
await driver.url('https://google.com')

await newDriver = attach(driver)
console.log(await newDriver.getUrl()) // outputs "Google"
```

Reply:
As far I'm aware **webdriver** package is not aware of the drivers/browsers constructed by the (multi)remote functions in the **webdriverio** package.

Probably in a separate pull request the webdriverio `attach` function could be refactored to accept a driver/browser instance which is constructed by `remote` or `multiremote`.
Even then, you still would preferably validate the (required) configuration here in `WebDriver.attachToSession`.

Reply:
Let's make changes to both methods then. One, in Webdriver so that we can do:
```js
const client = await WebDriver.newSession({
  path: '/',
  capabilities: { browserName: 'firefox' }
})
await client.navigateTo('https://www.google.com/ncr')

const client2 = await WebDriver.attachToSession(client)
console.log(await client2.getUrl()) // outputs "Google"
```
and make changes to the webdriverio attach method to be able to do the above example.

Reply:
I'm not sure whether I understand what you are trying to suggest in the previous comment, isn't `client` and `client2` the exact same?
There is no need to create `client2`, as you already have `client`. Both are single instances pointing to the same browser session, I think you are confusing attaching of sessions with adding of instances to `MultiRemoteDriver`.

Reply:
It's about attaching a (initial) session response from chrome/gecko/safari driver to WebDriver instance. Not attaching existing WebDriver instance to WebDriver instance, as you suggested in the previous `client`/`client2` example.

Reply:
why would you prefer to create a session manually instead of using the `remote` method?

Reply:
Isn't that the purpose of `attachToSession`?

Reply:
@martomo no, it is to attach to an existing webdriver session .. but there is no reason to initiate the session manually when you can use the `remote` method.